### PR TITLE
Add FXIOS-9059 [Microsurvey] Redux implementation for survey

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9059)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :bulb: Description
Add redux implementation for survey
- Dismiss survey + prompt
- Navigate to privacy policy page
- Submit survey to trigger telemetry

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

## Screenshots
https://github.com/mozilla-mobile/firefox-ios/assets/6743397/586b95f8-c6a2-40f9-8eee-2391a7ad8e80

### CODE REVIEW

**Review by OrlaM — APPROVED** [review]
LGTM so far, will take another look once it's out of draft.

You don't need to break this one up but in future try to keep your PR's small. Sometimes a large one is unavoidable but this one could have been a few eg:
- The set up code (subscribing to redux, the initial state etc)
- The coordinator logic
- Each use case could be it's own PR (close, submit, tab notice)
It doesn't need to be that many PRs just some suggestions for some of the boundaries it could be broken up along.

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 1]
```diff
@@ -594,8 +594,9 @@ class BrowserViewController: UIViewController,
                 guard microsurvey != nil else { return }
                 removeMicrosurveyPrompt()
             } else if state.microsurveyState.showSurvey {
-                // TODO: FXIOS-8895: Create Microsurvey Modal View
-                print("CYN - FXIOS-8895: Create Microsurvey Modal View")
+                // CYN: Torn between passing model and initializing the mobile messaging in the middleware
+                guard let model = state.microsurveyState.model else { return }
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:598]: add logging

**Comment by cyndichin** [comment]
Middleware Tests will be added in a separate PR

**Comment by cyndichin** [comment]
@DanielDervishi @tusharC95 this PR is related to the microsurvey demo from the grooming meeting. Feel free to let me know if you have any questions. This PR should have been broken up, so its a bit big, but let me know if you would like a live walkthrough to clarify anything!

**Review by OrlaM — APPROVED** [review]

**Comment by isabelrios** [comment]
Smoketest3 had a flaky test that passed in a re-try, general bitrise is ✅

**Pull request merged**